delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -724,7 +724,12 @@ genericConstraint            : typeReference
                              | CONSTRUCTOR
                              ;
 genericArguments             : '<' typeReferenceOrStringOrFile (',' typeReferenceOrStringOrFile)* '>'
-                             -> ^(TkGenericArguments<GenericArgumentsNodeImpl> typeReferenceOrStringOrFile typeReferenceOrStringOrFile*)
+                             -> ^(TkGenericArguments<GenericArgumentsNodeImpl> '<' typeReferenceOrStringOrFile (',' typeReferenceOrStringOrFile)* '>')
+                             ;
+genericDeclarationArguments  : '<' typeReferenceOrStringOrFile (commaOrSemicolon typeReferenceOrStringOrFile)* '>'
```

Comment:
I think `genericDeclarationArguments` is a bit of a misleading term. They are generic arguments for the *implementation*, not the original declaration. Perhaps `genericRoutineImplementationArguments` or similar?

Reply:
That's a fair point, I'm thinking `routineNameGenericArguments`.